File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonArrayProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/22/13 9:11 PM
 */
public class JsonArrayProperty extends JsonProperty {
    public JsonArrayProperty(String name, int id) {
        super(name, id);
    }

    public JsonArrayReader readValue(JsonObjectReader objectReader) {
        return (JsonArrayReader) objectReader.readPropertyValue();
    }

    public JsonArrayWriter write(JsonObjectWriter objectWriter) {
        return objectWriter.writeArrayProperty(this);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonArrayProperty.cs
namespace jsimple.json.readerwriter {

    /// <summary>
    /// @author Bret Johnson
    /// @since 5/22/13 9:11 PM
    /// </summary>
    public class JsonArrayProperty : JsonProperty {
        public JsonArrayProperty(string name, int id) : base(name, id) {
        }

        public virtual JsonArrayReader readValue(JsonObjectReader objectReader) {
            return (JsonArrayReader) objectReader.readPropertyValue();
        }

        public virtual JsonArrayWriter write(JsonObjectWriter objectWriter) {
            return objectWriter.writeArrayProperty(this);
        }
    }

}